drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,862 @@
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_MOD:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set mod %s\r\n", MODULATIONS[*(enum rn2483_mod_e *)data]);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_FREQ:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set freq %lu\r\n", *(uint32_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_BW:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set bw %hu\r\n", *(uint16_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_PRLEN:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set prlen %hu\r\n", *(uint16_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_CRC:
+  {
+    char *crc = (*(bool *)data) ? "on" : "off";
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set crc %s\r\n", crc);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_IQI:
+  {
+    char *iqi = (*(bool *)data) ? "on" : "off";
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set iqi %s\r\n", iqi);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_PWR:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set pwr %d\r\n", *(int8_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_SF:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set sf sf%d\r\n", *(uint8_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  case RN2483_SET_SYNC:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set sync %d\r\n", *(uint8_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  default:
+  {
+    // Not a valid set command
+    syslog(LOG_INFO, "INVALID COMMAND GIVEN %d", cmd);
+    return -EINVAL;
+  }
+  }
+
+  ssize_t write_length = file_write(&priv->uart, write_buffer, written);
+  if (write_length < 0)
+  {
+    return write_length;
+  }
+
+  // Read
+  int buf_len = 50;
+  char read_buffer[buf_len];
+  size_t read = 0;
+  ssize_t fread;
+  char last = '\0';
+
+  // Read data into buffer until \r\n terminating sequence reached
+  while (read < buf_len)
```

Comment:
I think you should extract this into another function like: `static int rn2483_read_response(char *buf, size_t nbytes)` which reads until the `\r\n` character in the response from the radio. You'll need this logic in the rest of the functions (read, write, ioctl).